src/components/course/CourseRecommendationCard.jsx

```diff
@@ -0,0 +1,116 @@
+import React, { useContext, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import Truncate from 'react-truncate';
+import { Link } from 'react-router-dom';
+import { AppContext } from '@edx/frontend-platform/react';
+import { Card } from '@edx/paragon';
+import { sendEnterpriseTrackEvent } from '@edx/frontend-enterprise-utils';
+
+import { isDefinedAndNotNull } from '../../utils/common';
+
+const CourseRecommendationCard = ({ course, isPartnerRecommendation }) => {
+  const { enterpriseConfig: { slug, uuid } } = useContext(AppContext);
+
+  const linkToCourse = useMemo(
+    () => {
+      if (!Object.keys(course).length) {
+        return '#';
+      }
+      return `/${slug}/course/${course.key}`;
+    },
+    [course],
+  );
+
+  const partnerDetails = useMemo(
+    () => {
+      if (!Object.keys(course).length || !isDefinedAndNotNull(course.owners)) {
+        return {};
+      }
+
+      return {
+        primaryPartner: course.owners?.length > 0 ? course.owners[0] : undefined,
+        showPartnerLogo: course.owners?.length === 1,
+      };
+    },
+    [course],
+  );
+
+  return (
+    <div
+      className="course-card-recommendation mb-4"
+      role="group"
+      aria-label={course.title}
+    >
+      <Link
+        to={linkToCourse}
+        onClick={() => {
+          let eventName = 'edx.ui.enterprise.learner_portal.recommended.course.clicked';
```

Comment:
Can we please put this logic outside for better clarity?

Reply:
Sure